waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/compatibility/HiveCompatibleThriftHiveMetastoreIfaceFactory.java

```diff
@@ -57,43 +57,36 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
           } else {
             throw compatibilityException.getCause();
           }
+        } catch (NoSuchMethodException e) {
```

Comment:
Rather than catch, log, and re-throw, why not wrap the exception with a new exception that contains the missing method name and let the callee decide what to do with it? (e.g., handle it, let it bubble, log it, etc.)